proto/secret/v1/secret.proto

```diff
@@ -0,0 +1,61 @@
+syntax = "proto3";
+package nitric.secret.v1;
+
+//protoc plugin options for code generation
+option go_package = "nitric/v1;v1";
+option java_package = "io.nitric.proto.secret.v1";
+option java_multiple_files = true;
+option java_outer_classname = "Secrets";
+option php_namespace = "Nitric\\Proto\\Secret\\V1";
+option csharp_namespace = "Nitric.Proto.Secret.v1";
+
+// The Nitric Secret Service contract
+service SecretService {
+  //Updates a secret, creating a new one if it doesn't already exist
+  rpc Put (SecretPutRequest) returns (SecretPutResponse);
+  //Gets a secret from a Secret Store
+  rpc Get (SecretGetRequest) returns (SecretGetResponse);
+}
+
+// Request to put a secret to a Secret Store
+message SecretPutRequest {
+  //The Secret to put to the Secret store
+  Secret secret = 1;
+}
+
+// Result from putting the secret to a Secret Store
+message SecretPutResponse {
+  //AWS returns ARN, name and VersionID
+  //GCP returns 
+  //Azure returns nothing
+
+  //The id of the secret
+  string name = 1;
```

Comment:
Since name and version id are in both PutResponse and GetRequest, should this be its own message type, like SecretIdentifier?

Reply:
There's also a separation of newly created secrets and accessed secrets, as each call the `Put` will create a new SecretVersion, that cannot be specified as part of the put request.

> NOTE: names below are only to highlight different cases and are not actual suggestions

```proto
// Fully qualified secret identifier
message SecretId {
  string name = 1;
  string version = 2;
}

// Essentially the SecretPutRequest
message NewSecret {
  // Use secret Id here as well, or not as its only half an ID
  string name = 1;
  bytes content = 2;
}

message ExistingSecret {
  SecretId id = 1;
  bytes content = 2;
}
```

Reply:
I've added a new message type SecretId which the GetRequest and PutResponse messages now use instead of the individual name, version_id fields.

``` proto
message SecretId {
  string name = 1;
  string version_id = 2;
}
```